iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/DataRegion.java

```diff
@@ -1105,18 +1103,15 @@ private void tryToUpdateInsertTabletLastCache(InsertTabletNode node) {
             latestFlushedTime);
   }
 
-  private void insertToTsFileProcessor(
-      InsertRowNode insertRowNode,
-      boolean sequence,
-      long timePartitionId,
-      Map<TsFileProcessor, Boolean> tsFileProcessorMapForFlushing)
+  private TsFileProcessor insertToTsFileProcessor(
+      InsertRowNode insertRowNode, boolean sequence, long timePartitionId)
       throws WriteProcessException {
     if (insertRowNode.allMeasurementFailed()) {
-      return;
+      return null;
     }
     TsFileProcessor tsFileProcessor = getOrCreateTsFileProcessor(timePartitionId, sequence);
     if (tsFileProcessor == null) {
-      return;
+      return null;
     }
```

Comment:
Maybe we can use `if (tsFileProcessor == null || insertRowNode.allMeasurementFailed())` to join two if clauses.

Reply:
fixed